### ISSUE TITLE
Can I make mutation while  no internet connection?

### ISSUE THREAD

**Issue by enggazzar** [apollographql/apollo-kotlin#5548]
### Question

I need to add items to the cart while there is no internet connection and when the mobile has internet mutation executed

**Comment by martinbonnin** [comment]
There is no such thing at the moment. You can [mutate your cache locally](https://www.apollographql.com/docs/kotlin/caching/store) but if you need that mutation to reach your server, you need to save it using something like [`WorkManager`](https://developer.android.com/topic/libraries/architecture/workmanager/)

**Comment by martinbonnin** [comment]
@enggazzar anything else we can help term here?

**Comment by martinbonnin** [comment]
Yes, that's correct, the `ApolloStore` API allows you to write data to the store offline and you can synchronize that later. 

If you're looking at a more concrete example, you can take a look at our benchmarks. They use `writeOperation` [here](https://github.com/apollographql/apollo-kotlin/blob/b4fb9fb7dea17b443fdce0fddce389bfd2331796/benchmark/app/src/main/kotlin/app.under.test/MainActivity.kt#L106)

**Comment by enggazzar** [comment]
writeOperation  is this mean execute mutation offline?

**Comment by martinbonnin** [comment]
`writeOperation` updates the contents of your local, offline, cache with the `data` that you are passing as argument. 

Most of the time, you don't need a mutation to write your local cache: create a `Query.data` instance programmatically and pass it to `writeOperation`. In your example, something like `writeOperation(cartContentsQuery, CartsContentQuery(items = listOf(newItemHere)), customScalarAdapters)`.

There's no silver bullet though and if you want that cart content to be synchronized to the backend, you'll have to execute the mutation later on.

**Comment by martinbonnin** [comment]
Hi 👋 I'll close this one in favour of https://github.com/apollographql/apollo-kotlin/issues/5618. That's something that would be nice to have although it's probably not for the short term.

**Comment by enggazzar** [comment]
Sure Thanks for your help
